src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -77,10 +77,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    
+    //here a regular expression is used to search for an image url entered by the user
+    //String regex is all the elements found in the url, it is the search pattern
+    // the link is then replaced with an img tag, a user is able to submit a
+    // pdf,png,jpg,gif,tiff, and bmp file
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    String regex = "(https?://\\S+\\.(png|jpg|gif|pdf|tiff|bmp))";
```

Comment:
That feature is already supported with this regular expression. It will take any embedded img addresses and just make them images. I'll make sure to fix my spaces and comment issue.